### PR TITLE
Update schedule UG description, edit jsonAdaptedPerson to get intended behaviour for json file.

### PR DESCRIPTION
**Pull request by geoffong11**
Add checking of `schedule` and `ScheduleName` to make sure they both exist or both do not. Update the `schedule` feature in UserGuide. Also trimmed all fields in jsonAdaptedPerson so that if someone edits the json file by adding whitespaces, it will still load the json file.

### CODE REVIEW

**Comment by geoffong11** [comment]
Closes #228

**Comment by geoffong11** [comment]
Closes #201 
Edit Phone number and relevant tests so that phone number length can only be 3 to 15 numbers long

File: src/main/java/connexion/storage/JsonAdaptedPerson.java [hunk 1]
```diff
@@ -163,31 +170,42 @@ public Person toModelType() throws IllegalValueException {
...
+        if (!ScheduleName.isValidScheduleName(trimmedScheduleName) && !trimmedScheduleName.isEmpty()) {
+            throw new IllegalValueException(ScheduleName.MESSAGE_CONSTRAINTS);
+        }
 
-        if (scheduleName == null) {
-            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
-                    ScheduleName.class.getSimpleName()));
+        if (trimmedScheduleName.isEmpty() && !trimmedSchedule.isEmpty()) {
+            throw new IllegalValueException(Schedule.MESSAGE_NO_SCHEDULE_NAME);
         }
```
Comment (xavierpok) [inline src/main/java/connexion/storage/JsonAdaptedPerson.java:193]: I might be missing something, but won't this disallow the combination of no name but present schedule?

Is this intended behaviour, or is the intended behaviour only for no schedule name to be allowed only in the UI?

If it's intended behaviour, might need a clause in the updated UG to reflect that the combo of no name + schedule is valid only for UI entry and NOT for manual JSON editing

File: src/test/java/connexion/storage/JsonAdaptedPersonTest.java [hunk 2]
```diff
@@ -205,6 +207,26 @@ public void toModelType_invalidScheduleName_throwsIllegalValueException() {
...
+    @Test
+    public void toModelType_blankScheduleNotBlankScheduleName_throwsIllegalValueException() {
+        JsonAdaptedPerson person =
+                new JsonAdaptedPerson(VALID_NAME, VALID_PHONE,
+                        VALID_EMAIL, VALID_COMPANY, VALID_JOB, VALID_TAGS, VALID_MARK_STATUS,
+                        BLANK_SCHEDULE, VALID_SCHEDULE_NAME, VALID_LAST_MODIFIED_DATE_TIME, VALID_NOTE);
+        String expectedMessage = ScheduleName.MESSAGE_NO_SCHEDULE;
+        assertThrows(IllegalValueException.class, expectedMessage, person::toModelType);
+    }
+
```
Comment (xavierpok) [inline src/test/java/connexion/storage/JsonAdaptedPersonTest.java:229]: A bit of a nit, but it might be better to add a test case testing the new whitespace stripping behaviour

Because that's arguably a new behaviour (in that there's some post-processing going on on the JSON input)

File: docs/UserGuide.md [hunk 3]
```diff
@@ -185,8 +185,10 @@ Format: `schedule INDEX i/SCHEDULE_TIME [a/SCHEDULE_NAME]`
...
 * Schedule name is an optional field. If no schedule name is given, the default is `Meeting`.
 * Input schedule time must be in the format `YYYY-MM-DD-HH-mm`, and must be valid.
+* Schedule times that are past today's date are allowed to be entered.
 * If there are existing schedules or schedule names, it will be updated to the input schedule and schedule name. If schedule name is not given, it will still be set to `Meeting`.
-* Updates the last modified time of the given person to the current time when executed
+* Only 1 schedule per person is allowed.
+* Updates the last modified time of the given person to the current time when executed.
 Examples:
 *  `schedule 1 i/2023-12-07-13-45` edits or adds the 1st person's schedule time and name, where the schedule time is `7 Dec 2023, 13:45:00`, and the schedule name is the default name, `Meeting`.
 *  `schedule 3 i/2024-05-06-18-00 a/Evening seminar` edits or adds the 3rd person's schedule time and name, where the schedule time is `6 May 2024, 18:00:00`, and the schedule name is `Evening seminar`.
```
Comment (xavierpok) [inline docs/UserGuide.md:194]: Another small documentation issue, we might need to specify the new behaviour of whitespace-stripping in the UG because it's not explicitly stated and hence ambiguous. The UI strips but to be safe maybe it will be best if we specify in the UG also?

I think it's ambiguous enough to go either way, so see what you think!

**Review by xavierpok — CHANGES_REQUESTED** [review]
LGTM for the most part, but I'm not too sure about the change to disallow the combo of no name + schedule time in the JSON file

**Comment by xavierpok** [comment]
> LGTM for the most part, but I'm not too sure about the change to disallow the combo of no name + schedule time in the JSON file

Overall, thanks for helping catch this issue that we missed!

File: docs/UserGuide.md (hunk 3 above)
Comment (xavierpok) [reply docs/UserGuide.md:194]: Non-critical, is ok to merge w/o addressing this

**Comment by geoffong11** [comment]
Feature flaw, not bug.
